File: Alkitab/src/yuku/alkitab/base/dialog/LabelEditorDialog.java
package yuku.alkitab.base.dialog;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.DialogInterface.OnClickListener;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup.LayoutParams;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.EditText;

import java.util.List;

import yuku.afw.V;
import yuku.alkitab.base.S;
import yuku.alkitab.base.model.Label;
import yuku.devoxx.flowlayout.R;

public class LabelEditorDialog {
	public static final String TAG = LabelEditorDialog.class.getSimpleName();
	
	public interface OkListener {
		void onOk(String judul);
	}
	
	public static void show(Context context, String initialText, String title, final OkListener okListener) {
		View dialogView = LayoutInflater.from(context).inflate(R.layout.dialog_edit_label, null);
		final EditText tJudul = V.get(dialogView, R.id.tJudul);
		tJudul.setText(initialText);
		
		final AlertDialog dialog = new AlertDialog.Builder(context)
		.setView(dialogView)
		.setTitle(title)
		.setPositiveButton(R.string.ok, new OnClickListener() {
			@Override public void onClick(DialogInterface dialog, int which) {
				if (okListener != null) {
					okListener.onOk(tJudul.getText().toString().trim());
				}
			}
		})
		.setNegativeButton(R.string.cancel, null)
		.create();
		
		dialog.getWindow().setLayout(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
		dialog.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
		dialog.show();
		
		final Button bOk = dialog.getButton(AlertDialog.BUTTON_POSITIVE);
		bOk.setEnabled(false);
		
		final List<Label> semuaLabel = S.getDb().listSemuaLabel();
		
		tJudul.addTextChangedListener(new TextWatcher() {
			@Override public void onTextChanged(CharSequence s, int start, int before, int count) {
			}
			
			@Override public void beforeTextChanged(CharSequence s, int start, int count, int after) {
			}
			
			@Override public void afterTextChanged(Editable s) {
				if (s.length() == 0 || s.toString().trim().length() == 0) {
					bOk.setEnabled(false);
					return;
				} else {
					String judulBaruTrim = s.toString().trim();
					for (Label label: semuaLabel) {
						if (label.judul.trim().equals(judulBaruTrim)) {
							bOk.setEnabled(false);
							return;
						}
					}
				}
				bOk.setEnabled(true);
			}
		});
	}
}

File: Alkitab/src/main/java/yuku/alkitab/base/dialog/LabelEditorDialog.kt
package yuku.alkitab.base.dialog

import android.content.Context
import android.content.DialogInterface
import android.text.Editable
import android.text.InputFilter
import android.text.InputType
import android.text.TextWatcher
import android.view.LayoutInflater
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.android.material.textfield.TextInputEditText
import com.google.android.material.textfield.TextInputLayout
import yuku.alkitab.base.S.db
import yuku.alkitab.debug.R

private const val MAX_LABEL_LENGTH = 48

object LabelEditorDialog {
    @JvmStatic
    fun show(context: Context, initialText: String, title: String, okListener: OkListener) {
        val allLabels = db.listAllLabels()

        val dialogView = LayoutInflater.from(context).inflate(R.layout.dialog_input, null, false)
        val til = dialogView.findViewById<TextInputLayout>(R.id.tilInput)
        val et = dialogView.findViewById<TextInputEditText>(R.id.etInput)

        til.hint = context.getString(R.string.nama_label_titikdua)
        til.isCounterEnabled = true
        til.counterMaxLength = MAX_LABEL_LENGTH
        et.inputType = InputType.TYPE_CLASS_TEXT
        et.filters = arrayOf(InputFilter.LengthFilter(MAX_LABEL_LENGTH))
        et.setText(initialText)

        val dialog = MaterialAlertDialogBuilder(context)
            .setTitle(title)
            .setView(dialogView)
            .setPositiveButton(R.string.ok) { _, _ ->
                okListener.onOk(et.text.toString().trim())
            }
            .setNegativeButton(R.string.cancel, null)
            .create()

        fun isValid(text: CharSequence?): Boolean {
            val trimmed = text?.toString()?.trim().orEmpty()
            return trimmed.isNotEmpty() && trimmed.length <= MAX_LABEL_LENGTH &&
                allLabels.none { it.title.trim() == trimmed }
        }

        et.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) = Unit
            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) = Unit
            override fun afterTextChanged(s: Editable?) {
                dialog.getButton(DialogInterface.BUTTON_POSITIVE)?.isEnabled = isValid(s)
            }
        })

        dialog.show()
        // Derive initial button state from initialText — e.g. a caller pre-filling a valid,
        // non-duplicate name should get an immediately-clickable OK.
        dialog.getButton(DialogInterface.BUTTON_POSITIVE)?.isEnabled = isValid(initialText)
    }

    fun interface OkListener {
        fun onOk(title: String?)
    }
}
